lib/src/configuration.dart

```diff
@@ -383,16 +383,64 @@ class Configuration {
     return pubspec;
   }
 
+  String? _getPubspecVersionWithBuildNumber(Version pubspecVersion) {
+    const prefix = "Warning: version_with_build_number is enabled,";
+    if (pubspecVersion.build.isNotEmpty && pubspecVersion.build[0] is int) {
+      final buildNumber = pubspecVersion.build[0] as int;
+      if (pubspecVersion.major > 65535) {
```

Comment:
Doesn't Microsoft require the major to be >0 as well? Although should probably put the test below since it affects the other auto-version behavior

Reply:
I have recently released an application on the windows store, with a `0` as the major version multiple times.
The screenshot below shows this. To verify it for yourself, you can search for `tivatrainer` in the windows store, or visit the following link:
https://apps.microsoft.com/store/detail/tivatrainer/9PC7N0B9ZPQH?hl=nl-nl&gl=nl&rtc=1
 
![windows_listing](https://github.com/YehudaKremer/msix/assets/16548374/c8e2818a-0334-4bcc-b3bd-f0f3a9641750)

Reply:
Yeah, I was just looking at the docs. If it works and doesn't seem like Microsoft will change it, then no worries. I thought it a bit odd to limit `0.x` versions as well for that reason.

Side note: that was one of the reasons for combining major and minor versions into the first field as it would remove that limitation. I think minor+patch is better overall, though.